Harmony/Public/Patch.cs

```diff
@@ -5,13 +5,31 @@
 using System.Reflection.Emit;
 using System.Runtime.Serialization;
 using System.Runtime.Serialization.Formatters.Binary;
+#if NET50_OR_GREATER
+using System.Runtime;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+#endif
 
 namespace HarmonyLib
 {
 	/// <summary>Patch serialization</summary>
-	/// 
+	///
 	internal static class PatchInfoSerialization
 	{
+#if NET50_OR_GREATER
+		static bool useBinaryFormatter
+		{
+			get
+			{
+				// https://github.com/dotnet/runtime/blob/208e377a5329ad6eb1db5e5fb9d4590fa50beadd/src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/LocalAppContextSwitches.cs#L14
+				bool hasSwitch = AppContext.TryGetSwitch("System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization", out bool isEnabled);
```

Comment:
Can this be expected to change during runtime or can we put it into a static bool?

Reply:
It looks like .NET caches it, so we can do the same